functions/pupil-prefs/index.js

```diff
@@ -49,19 +69,18 @@ module.exports = async function (context, pupilPrefsMessage) {
  * @param {String} prefTable - reference table to match id
  * @param {String} prefCode - code value to be matched with reference table
  * @param {String} accessArrangementCode - code value for relevant access arrangement
- * @param {Function} logger
  * @return {Promise<void>}
  */
-async function updatePupilAccessArrangementsPreference (checkCode, prefField, prefTable, prefCode, accessArrangementCode, logger) {
+async function updatePupilAccessArrangementsPreference (checkCode, prefField, prefTable, prefCode, accessArrangementCode) {
   const sql = `UPDATE ${schema}.${pupilAccessArrangementsTable}
                SET ${prefField} = (
                 SELECT id FROM ${schema}.${prefTable} 
                 WHERE code = @prefCode
                )                
                WHERE pupil_Id = (
                   SELECT p.id FROM ${schema}.${pupilTable} p
-                  LEFT JOIN ${schema}.${checkTable} chk
-                  ON chk.pupil_id= p.id
+                  FULL OUTER JOIN ${schema}.${checkTable} chk

```

Comment:
please check to ensure you really do want all rows from both tables.  i would expect not